src/Microsoft.ML.Core/Prediction/TrainContext.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Data;
+
+namespace Microsoft.ML.Runtime
+{
+    /// <summary>
+    /// Instances of this class are meant to be constructed and passed to trainers.
```

Comment:
Insufficient documentation. While the easy case (no validation, no init predictor) is "easy," how to deal with those complexities for those that find it necessary will be important... this includes both users and implementors of `ITrainer`.